### PR TITLE
Convert to v2 addon

### PR DESCRIPTION
**Pull request by SergeAstapov**
Follows [Porting an Addon to V2](https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md) guide

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "docs",
+  "version": "2.0.0-beta.3",
+  "private": true,
+  "description": "Docs app for ember-cli-page-object",
+  "scripts": {
+    "build": "node ./index.js"
```
Comment (SergeAstapov) [inline docs/package.json:7]: now docs/guides completely separated from the addon and the test addon

File: test-app/.eslintrc.js [hunk 2]
```diff
@@ -2,12 +2,17 @@
 
 module.exports = {
   root: true,
+  parser: '@babel/eslint-parser',
```
Comment (SergeAstapov) [inline test-app/.eslintrc.js:5]: this allows using ESLint v8

**Comment by ro0gr** [comment]
Thank you very much for this work! I'll do my best to check it this weekend.

**Comment by ro0gr** [comment]
Overall looks fantastic!

Me as a not active monorepo user is mostly interested is how to release the thing? At the moment we have the following release instructions https://github.com/san650/ember-cli-page-object/wiki/How-to-release-a-new-version.

May you provide steps to release the `/addon` with `/docs`, so I can figure out how to update documentation, please?

**Comment by NullVoxPopuli** [comment]
any chance we can get this released -- I'm trying to upgrade to ember 4 / deal with ember 3.28 deprecations and ember-cli-page-object is my last blocker

----------


@ro0gr , re: releasing

the addon folder with "just" be `npm publish` (after `cd`ing into `/addon`)

for docs, looks like you use this: https://github.com/san650/ember-cli-page-object/blob/master/docs.js

we'd need to test it out, but I think you'd move that file into the docs folder and just invoke it.
oh! this needs updating: https://github.com/san650/ember-cli-page-object/blob/master/docs.js#L264

**Comment by NullVoxPopuli** [comment]
@SergeAstapov is this a non-breaking change? (other than now requiring ember-auto-import@v2?)

**Comment by SergeAstapov** [comment]
@ro0gr I'm sorry, missed you question.

TL;DR I will create a PR adding release-it config and adding `RELEASE.md` with release instructions (which would be very simple).

Longer answer.
I would go with `release-it` for release automation: it is pretty simple to setup (e.g. you can see [PR in ember-click-outside-modifier](https://github.com/lifeart/ember-click-outside-modifier/pull/16/files) as an example) and we can setup docs deploy via the hook (like it's done in [`ember-keyboard`](https://github.com/adopted-ember-addons/ember-keyboard/blob/master/docs/package.json#L20)).

As a pros, it automatically publishes to npm, creates GitHub release, generates changelog and monorepo setup is super simple (thanks to [release-it-yarn-workspaces](https://github.com/rwjblue/release-it-yarn-workspaces) plugin)

The only downside (in my opinion) - it requires labels to be added to pull requests.

semantic-release or standard-version could be another option, however to me they are not that straightforward and easy to use as `release-it` *and* they require diligence in commit messages (which hard to change retrospectively compared to PR labels)

<hr>
@NullVoxPopuli correct, no breaking changes except ember-auto-import@v2 (which *is* a breaking change).

File: docs/index.js [hunk 3]
```diff
@@ -18,6 +18,8 @@ const RSVP = require('rsvp');
 const packageJSON = require('./package.json');
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply docs/index.js:18]: only ember-cli-page-object needs its version changed?
or am I missing something? why would we bump the other packages? (they can stay at 0.0.0 (or whatever they are), yea?)

[the remark this answers is not in the document]
Comment (SergeAstapov) [reply docs/index.js]: @ro0gr thanks for pointing this out! indeed this was a miss on my side and now is fixed

**Comment by SergeAstapov** [comment]
@ro0gr sure, totally makes sense to keep release process as is and I'll submit follow up PR where we can work on automation separately.

We just should make following changes in https://github.com/san650/ember-cli-page-object/wiki/How-to-release-a-new-version:

* make sure both `npm version` and `npm publish` run from `addon` folder.
* replace `npm run docs` command with `cd ../docs && npm run build`

Rest of the steps don't seem to change.

**Comment by ro0gr** [comment]
Alright, seems like downgrading to the "@ember/test-helpers": "~2.7.0" does the trick. For some reason we end up in the https://github.com/emberjs/ember-test-helpers/blob/2b1ddcc91a971e1a3e1509d55f711778c32984b1/addon-test-support/%40ember/test-helpers/-internal/get-component-manager.ts#L14-L17, which does not make sense to me and might mean some underlying bug there. I'll need some time to wrap my head around this issue. 

Meanwhile feel free to add a commit to downgrade the test helpers if you want for now. Anyways, I'll do my best to finally cut the release this week.

File: test-app/package.json [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    "coveralls": "^3.0.0",
+    "ember-auto-import": "^2.4.1",
+    "ember-cli": "~4.0.0",
+    "ember-cli-babel": "^7.26.11",
+    "ember-cli-code-coverage": "1.0.0-beta.3",
+    "ember-cli-dependency-checker": "^3.3.1",
+    "ember-cli-doc-server": "1.1.0",
+    "ember-cli-htmlbars": "^6.0.1",
+    "ember-cli-inject-live-reload": "^1.8.2",
+    "ember-cli-page-object": "2.0.0-beta.3",
```
Comment (ro0gr) [inline test-app/package.json:73]: I've just noticed this, and I wonder if this is supposed to be a relative path, or use `*` instead of the hardcoded version?

**Comment by ro0gr** [comment]
Well, I figured out test failures and fixed it by migrating to pnpm https://github.com/san650/ember-cli-page-object/pull/579

**Comment by ro0gr** [comment]
@SergeAstapov thank you! 🚀

**Comment by NullVoxPopuli** [comment]
> fixed it by migrating to pnpm 

Yet another situation where `pnpm` saves the day!

**Comment by ro0gr** [comment]
just published it as [2.0.0-beta.4](https://github.com/san650/ember-cli-page-object/releases/tag/v2.0.0-beta.4)
